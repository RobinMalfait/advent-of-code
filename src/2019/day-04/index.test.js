import { describe, expect, it } from 'vitest'
import part1 from './part-1'
import part2 from './part-2'

describe('Part 1', () => {
  // Part 1
  it.each([
    ['111111', true],
    ['223450', false],
    ['123789', false],
  ])('An input of %s requires an output of "%s"', (input, expected) => {
    expect(part1.isValidPassword(input)).toBe(expected)
  })

  // Actual test, Part 1
  it('should produce the correct value for the input data', async () => {
    expect(part1.countValidPasswordsBetween('234208', '765869')).toMatchSnapshot()
  })
})

describe('Part 2', () => {
  // Part 2
  it.each([
    ['112233', true],
    ['123444', false],
    ['111122', true],
  ])('An input of %s requires an output of "%s"', (input, expected) => {
    expect(part2.isValidPassword(input)).toBe(expected)
  })

  // Actual test, Part 2
  it('should produce the correct value for the input data', async () => {
    expect(part2.countValidPasswordsBetween('234208', '765869')).toMatchSnapshot()
  })
})
