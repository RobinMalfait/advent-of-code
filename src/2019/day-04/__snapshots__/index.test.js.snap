// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Part 1 should produce the correct value for the input data 1`] = `1246`;

exports[`Part 2 should produce the correct value for the input data 1`] = `814`;
